tools/themis/src/rules.rs

```diff
@@ -0,0 +1,351 @@
+use std::collections::HashMap;
+
+use super::{style, utils, Error, Expected, PackageOutcome, Workspace};
+
+/// ensure all crates have the `publish = <true/false>` specification
+pub fn has_publish_spec(workspace: &Workspace) -> Result<(), Error> {
+    let outliers: Vec<_> = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.raw["package"].get("publish").is_none())
+        .map(|pkg| PackageOutcome { pkg, value: None })
+        .collect();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages should have the `publish` specification".to_string(),
+            expected: Some(Expected { value: "publish = <true/false>".to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    return Ok(());
```

Comment:
```suggestion
    Ok(())
```

Reply:
Yeah, this is one of the refactorings I was going to get to.